File: cpp/example/src/PathSum/pathSum.cpp
#include "pathSum.h"

bool pathSum::naive(TreeNode* root,int targetSum){
    if (root == nullptr) return false;
    if (root->left == nullptr && root->right == nullptr)
        return root->val == targetSum;
    bool result = false;
    if (root->left != nullptr && pathSum::naive(root->left,targetSum-root->val))
        result = true;
    if (root->right != nullptr && pathSum::naive(root->right,targetSum-root->val))
        result = true;
    return result;
}

File: cs/sample/Sample/PathSum/PathSum.cs
using System;
using System.Collections.Generic;
using Sample.TreeNode;

namespace Sample.PathSum{
    public class pathSum{
        public static bool naive(treeNode root,int targetSum){
            if (root == null) return false;
            if (root.left == null && root.right == null)
                return root.val == targetSum;
            bool result = false;
            if (root.left != null && pathSum.naive(root.left,targetSum-root.val))
                result = true;
            if (root.left != null && pathSum.naive(root.right,targetSum-root.val))
                result = true;
            return result;
        }
    }
}
